jung-samples/src/main/java/edu/uci/ics/jung/samples/AddNodeDemo.java

```diff
@@ -81,25 +83,24 @@ public void handleGraphEvent(NetworkEvent<Number, Number> evt) {
               case EDGE_REMOVED:
                 NetworkEvent.Edge<Number, Number> edgeEvent =
                     (NetworkEvent.Edge<Number, Number>) evt;
-                System.err.println(
-                    "got "
-                        + evt
-                        + " with endpoints: "
-                        + original.incidentNodes(edgeEvent.getEdge()));
+                if (log.isDebugEnabled()) {
+                  log.debug(
+                      "got {} with endpoints:{}", evt, original.incidentNodes(edgeEvent.getEdge()));
+                }
```

Comment:
I believe that when @Stephan202 commented earlier about our usage of SLF4J, he meant we can remove `if (log.isDebugEnabled()) { [...] }` wherever we're currently using it now, as the most expensive operation in logging - string concatenation - no longer happens eagerly.

Reply:
I didn't mean it, but I totally agree with it ;)

Reply:
;)

Reply:
But having said this, we do still call operations like `Graph.incidentNodes([...])`, which can be expensive, so I have my doubts now...

Reply:
After the code visualization works again, I will probably go through and remove pointless logging and add more logging that would benefit users.
All this focus on logging and formatting now is a bit like fixing the hole in the Titanic and having someone say that the paint does not quite match.

Reply:
Agreed. So I'd be more than happy to keep my hands off the proverbial paint, and wait for @jrtom's review now. :)

Reply:
Your comments made the code better. That matters a lot. I appreciate your trying the demos to see if anything is still misbehaving.